mcumgr-core/src/main/java/io/runtime/mcumgr/managers/ImageManager.java

```diff
@@ -627,6 +636,23 @@ public synchronized boolean upload(byte @NotNull [] data, @NotNull ImageUploadCa
         mUploadCallback = callback;
         mImageData = data;
 
+        byte[] dataHash = null;
+
+        /*
+         * Feature in Apache Mynewt: Device keeps track of unfinished uploads based on the
+         * SHA256 hash over the image data. When an upload request is received which contains
+         * the same hash of a partially finished upload, the device will send the offset to
+         * continue from. The hash is truncated to save packet
+         */
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            dataHash = digest.digest(data);
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+
+        mImageDataHash = dataHash;
```

Comment:
If I've read correctly, this method is deprecated, hence the duplication here.
Happy to move this if it lives better somewhere else?